From 120a6b77c03842cd0de9450916600690a8b7e300 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Thu, 11 Jul 2019 20:11:49 +0200
Subject: [PATCH 16/23] gtkwindow: Maximize dialogs when mapping

That is when initially maximized windows are expected to be maximized,
not just before realizing them.
---
 gtk/gtkwindow.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index dc561af088..395c9dfdce 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -6263,6 +6263,7 @@ gtk_window_map (GtkWidget *widget)
   GdkWindow *gdk_window;
   GList *link;
   GdkDisplay *display;
+  gboolean is_dialog;
 
   if (!_gtk_widget_is_toplevel (widget))
     {
@@ -6287,7 +6288,10 @@ gtk_window_map (GtkWidget *widget)
 
   gdk_window = _gtk_widget_get_window (widget);
 
-  if (priv->maximize_initially)
+  /* Transient windows can be considered pseudo-dialogs. */
+  is_dialog = GTK_IS_DIALOG (widget) || !!gtk_window_get_transient_for (window);
+  if (priv->maximize_initially ||
+      (is_dialog && gtk_window_get_resizable (window)))
     gdk_window_maximize (gdk_window);
   else
     gdk_window_unmaximize (gdk_window);
@@ -7362,16 +7366,10 @@ gtk_window_realize (GtkWidget *widget)
   GtkWindowPrivate *priv;
   gint i;
   GList *link;
-  gboolean is_pseudo_dialog;
 
   window = GTK_WINDOW (widget);
   priv = window->priv;
 
-  is_pseudo_dialog = !GTK_IS_DIALOG (widget) &&
-                     !!gtk_window_get_transient_for (window);
-  if (is_pseudo_dialog && gtk_window_get_resizable (window))
-    gtk_window_maximize (window);
-
   if (!priv->client_decorated && gtk_window_should_use_csd (window))
     create_decoration (widget);
 
-- 
2.28.0

